app/main.py

```diff
@@ -1,25 +1,62 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Any, owner: Any) -> Any:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError(f"{value} should be integer!")
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"{value} should be between {self.min_amount} and {self.max_amount}")
+        setattr(instance, self.protected_name, value)
```

Comment:
The `IntegerRange` class is well implemented. It might be useful to add a docstring to describe its purpose and how it works for future reference.